src/createTwitchVideoCompiler/youtubeUploader.js

```diff
@@ -0,0 +1,192 @@
+// @flow
+
+import fs from 'fs';
+import readline from 'readline';
+import { google } from 'googleapis';
+
+export default async function youtubeUploader(
+  clientFilePath: string,
+  properties: {
+    language: string,
+    description: string,
+    tags: string,
+    title: string,
+    privacy: string,
+    videoPath: string,
+  },
+): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const { OAuth2 } = google.auth;
+    const SCOPES = ['https://www.googleapis.com/auth/youtube.force-ssl'];
+    // $FlowFixMe
+    const TOKEN_DIR = `${process.env.HOME ||
+      // $FlowFixMe
+      process.env.HOMEPATH ||
+      // $FlowFixMe
+      process.env.USERPROFILE}/.credentials/`;
+    const TOKEN_PATH = `${TOKEN_DIR}twitch-parser.json`;
+
+    function storeToken(token) {
+      try {
+        fs.mkdirSync(TOKEN_DIR);
+      } catch (err) {
+        if (err.code !== 'EEXIST') {
+          throw err;
+        }
+      }
+      fs.writeFile(TOKEN_PATH, JSON.stringify(token));
+      console.log(`Token stored to ${TOKEN_PATH}`);
+    }
+
+    function getNewToken(oauth2Client, requestData, callback) {
+      const authUrl = oauth2Client.generateAuthUrl({
+        access_type: 'offline',
+        scope: SCOPES,
+      });
+
+      console.log('Authorize this app by visiting this url: ', authUrl);
+
+      const rl = readline.createInterface({
```

Comment:
You should move this outside of `youtubeUploader()`. One reason is we want to run this programmatically in the future and we can't have this function to stop and wait for user input.

Another reason is this: imagine you want to generate a compilation of hundred videos. You start the process and go to get some coffee. When you come back, expecting the tool to be done, you will end up seeing a prompt asking for more input. This is bad UX. Just ask everything you have to ask first, and then start the process.